src/helper/mutations.ts

```diff
@@ -228,44 +159,108 @@ export function useCreateThreadMutation() {
-        mutation createComment($input: ThreadCreateCommentInput!) {
-          thread {
-            createComment(input: $input) {
-              success
-            }
-          }
-        }
-      `,
-      variables: { input },
-    }
-
-    try {
-      const response = await authFetchThread(args, auth)
+    const success = await mutationFetch(auth, mutation, input)
 
-      if (response.thread.createComment?.success) {
-        return !!(await mutate(`comments::${entityId}`))
-      }
-    } catch (e) {
-      handleError(showToastNotice, e)
-      return false
-    }
+    if (success) await mutate(`comments::${entityId}`)
+    return success
   }
 
   return async (input: ThreadCreateCommentInput) =>
     await createCommentMutation(input)
 }
 
-function handleError(
-  showToastNotice: ReturnType<typeof useToastNotice>,
-  response?: object
-) {
-  console.log(response)
-  showToastNotice(`Das hat leider nicht geklappt…`, 'warning')
+type MutationInput =
+  | NotificationSetStateInput
+  | UuidSetStateInput
+  | ThreadCreateThreadInput
+  | ThreadSetThreadArchivedInput
+  | ThreadSetThreadStateInput
+  | ThreadSetCommentStateInput
+  | ThreadCreateCommentInput
+
+type ApiErrorType = 'UNAUTHENTICATED' | 'FORBIDDEN'
+type ErrorType = ApiErrorType | 'UNKNOWN'
+
+export interface ApiError extends GraphQLError {
+  extensions: {
+    code: ApiErrorType
+  }
+}
+
+export async function mutationFetch(
+  auth: React.RefObject<AuthPayload>,
+  query: string,
+  input: MutationInput,
+  isRetry?: boolean
+): Promise<boolean> {
+  if (auth.current === null) return handleError('UNAUTHENTICATED')
+
+  try {
+    return await executeQuery()
+  } catch (e) {
+    const error = (e as ClientError).response.errors?.[0] as
+      | ApiError
+      | undefined
+
+    const type = error ? error.extensions.code : 'UNKNOWN'
+
+    if (type === 'UNAUTHENTICATED' && !isRetry) {
+      await auth.current.refreshToken()
+      return await mutationFetch(auth, query, input, true)
+    }
+
+    return handleError(type)
+  }
+
+  async function executeQuery(): Promise<boolean> {
+    const client = new GraphQLClient(endpoint, {
+      headers: auth.current
+        ? {
+            'access-control-allow-origin': '*',
+            Authorization: `Bearer ${auth.current.token}`,
+          }
+        : {},
+    })
+    // … okay for now
+    return (
+      JSON.stringify(await client.request(query, { input })).indexOf(
+        '"success":true'
+      ) > -1
+    )
+  }
+}
+
+function handleError(type: ErrorType, e?: object): false {
+  const message =
+    type == 'UNAUTHENTICATED'
+      ? 'Für diese Funktion musst du dich einloggen!'
+      : type == 'FORBIDDEN'
+      ? 'Dafür fehlen dir leider die Rechte!'
+      : 'Ein unbekannter Fehler…'
+
+  if (type == 'UNKNOWN') {
+    console.log(e)
+    triggerSentry({ message: 'Unknown API error' })
+  }
+
+  if (type == 'UNAUTHENTICATED') {
+    //make sure correct auth state is showing
+    csrReload()
```

Comment:
Ping #851. Not a fan of this hacks tbh, since they allow us to be sloppy about the state handling (instead of fixing the actual problem why this doesn't work right now).

Reply:
Okay, we leave it (because it's better than the current state) and try to find a fix in #851.